utils/utils.go

```diff
@@ -153,42 +207,83 @@ func CreateResultFromEndpoint(ep *api.WorkloadEndpoint) (*current.Result, error)
 	return result, nil
 }
 
-func GetIdentifiers(args *skel.CmdArgs) (workloadID string, orchestratorID string, err error) {
+// PopulateEndpointNets takes a WorkloadEndpoint and a CNI Result, extracts IP address and mask
+// and populates that information into the WorkloadEndpoint.
+func PopulateEndpointNets(wep *api.WorkloadEndpoint, result *current.Result) error {
+	if len(result.IPs) == 0 {
+		return errors.New("IPAM plugin did not return any IP addresses")
+	}
+
+	for _, ipNet := range result.IPs {
+		if ipNet.Version == "4" {
+			ipNet.Address.Mask = net.CIDRMask(32, 32)
+		} else {
+			ipNet.Address.Mask = net.CIDRMask(128, 128)
+		}
+
+		wep.Spec.IPNetworks = append(wep.Spec.IPNetworks, ipNet.Address.String())
+	}
+
+	return nil
+}
+
+type EndpointIdentifiers struct {
+	PodName        string
+	Namespace      string
+	ContainerID    string
+	InterfaceName  string
+	OrchestratorID string
+	WEPName        string
+	NodeName       string
+}
+
+// GetIdentifiers takes CNI command arguments, and extracts identifiers i.e. pod name, pod namespace,
+// container ID, and orchestratorID based on the orchestrator.
+func GetIdentifiers(args *skel.CmdArgs, nodename string) (*EndpointIdentifiers, error) {
 	// Determine if running under k8s by checking the CNI args
-	k8sArgs := K8sArgs{}
-	if err = types.LoadArgs(args.Args, &k8sArgs); err != nil {
-		return workloadID, orchestratorID, err
+	k8sArgs := types.K8sArgs{}
+	if err := cniTypes.LoadArgs(args.Args, &k8sArgs); err != nil {
+		return &EndpointIdentifiers{"", "", "", "", "", "", ""}, err
```

Comment:
You don't need to include the values like that.  Just use &EndpointIdentifiers{}.
In fact, it's a pointer, why can't you just return nil?